PhysicsTools/PatAlgos/plugins/PATPackedCandidateProducer.cc

```diff
@@ -320,7 +333,26 @@ void pat::PATPackedCandidateProducer::produce(edm::StreamID, edm::Event& iEvent,
 	} else {
 	  outPtrP->back().setHcalFraction(0);
 	}
+       
+	std::vector<float> dummyVector; dummyVector.clear();
+	pat::HcalDepthEnergyFractionProd hcalDepthEFrac(dummyVector);
 	
+	// storing HcalDepthEnergyFraction information
+	if ( std::find(pfCandidateTypesForHcalDepth_.begin(), pfCandidateTypesForHcalDepth_.end(), abs(cand.pdgId())) 
+	     != pfCandidateTypesForHcalDepth_.end() ){
+	  if (!storeHcalDepthEndcapOnly_ || fabs(outPtrP->back().eta())>1.3 ){  // storeHcalDepthEndcapOnly_==false -> store all eta of selected 
+	                                                                // PF types, if true, only |eta|>1.3 of selected PF types will be stored  
+	    std::vector<float> hcalDepthEnergyFractionTmp {
+	        cand.hcalDepthEnergyFraction(1), cand.hcalDepthEnergyFraction(2),
+	        cand.hcalDepthEnergyFraction(3), cand.hcalDepthEnergyFraction(4),
```

Comment:
this can also be ```std::vector<float> hcalDepthEnergyFractionTmp (cand.hcalDepthEnergyFractions().begin(), cand.hcalDepthEnergyFractions().end());```